addon/utils/promise-tools.js

```diff
@@ -0,0 +1,56 @@
+import Ember from 'ember';
+
+const isPromise = function(maybePromise) {
+  if (Ember.PromiseProxyMixin.detect(maybePromise)) {
+    return true;
+  }
+
+  if (maybePromise instanceof Ember.RSVP.Promise) {
+    return true;
+  }
+
+  if (maybePromise != null && typeof maybePromise.then === 'function') {
```

Comment:
true, mostly it was copied from other code. I'll check it to a then check only here